API_doc/LibConfig.hpp

```diff
@@ -0,0 +1,65 @@
+//
+// Must be included to use the LibConfig
+//
+//
+
+#pragma once
+
+# include <cstddef>
+
+/*
+ * Base class for all operations
+ */
+class LibConfig
+{
+
+public:
+
+  using onChangeNotifier = void (*)(void *data, char const *newValue);
+
+  /*
+   * contain a unique id for each configuration
+   */
+  struct Id {
+    void *data;
+    size_t size;
+  };
+
+  /*
+   * manage your subscriptions
+   */
+  class Subscription
+  {
+    onChangeNotifier subscribedFunc;
+  public:
+
+    Subscription(onChangeNotifier);
+
+    void unsubscribe(); // stop your subscription
+  };
+
+  /*
+   * Config class is used to manipulate configuration
+   */
+  class Config
+  {
+
+    Id configId;
+
+  public:
+
+    Id const &getId() const;
```

Comment:
This is inconsistent with `getConfig`, either an `Id` should be cheap to copy (in that case no ref) or it should be taken by reference. I would think it's cheap to copy?